test/Datadog.Trace.IntegrationTests/SendTracesToZipkinCollector.cs

```diff
@@ -7,41 +7,103 @@
 
 namespace Datadog.Trace.IntegrationTests
 {
-    public class SendTracesToZipkinCollector
+    public class SendTracesToZipkinCollector : IDisposable
     {
         private readonly Tracer _tracer;
-        private readonly int collectorPort = 9411;
+        private readonly MockZipkinCollector _zipkinCollector;
 
         public SendTracesToZipkinCollector()
         {
-            var agentUri = new Uri("http://localhost:9411/api/v2/spans");
+            int collectorPort = 9411;
+            var agentUri = new Uri($"http://localhost:{collectorPort}/api/v2/spans");
             var exporter = new ZipkinExporter(agentUri);
             var exporterWriter = new ExporterWriter(exporter, new NullMetrics());
             _tracer = new Tracer(new TracerSettings(), exporterWriter, sampler: null, scopeManager: null, statsd: null);
+            _zipkinCollector = new MockZipkinCollector(collectorPort);
+        }
+
+        public void Dispose()
+        {
+            _zipkinCollector.Dispose();
         }
 
         [Fact]
         public void MinimalSpan()
         {
-            using (var zipkinCollector = new MockZipkinCollector(collectorPort))
+            const string operationName = "MyOperation";
+            using (var scope = _tracer.StartActive(operationName))
             {
-                var scope = _tracer.StartActive("Operation");
                 scope.Span.SetTag(Tags.SpanKind, SpanKinds.Client);
                 scope.Span.SetTag("key", "value");
-                scope.Dispose();
+            }
 
-                var spans = zipkinCollector.WaitForSpans(1);
+            var zspan = RetrieveSpan();
 
-                Assert.Single(spans);
-                var zspan = spans[0];
-                Assert.Equal(scope.Span.OperationName, zspan.Name);
-                Assert.True(zspan.Tags.TryGetValue("key", out var tagValue));
-                Assert.Equal("value", tagValue);
+            Assert.Equal(operationName, zspan.Name);
+            Assert.True(zspan.Tags.TryGetValue("key", out var tagValue));
+            Assert.Equal("value", tagValue);
 
-                // The span.kind has an special treatment.
-                Assert.False(zspan.Tags.ContainsKey(Tags.SpanKind));
-                Assert.Equal("CLIENT", zspan.Kind);
+            // The span.kind has an special treatment.
+            Assert.False(zspan.Tags.ContainsKey(Tags.SpanKind));
+            Assert.Equal("CLIENT", zspan.Kind);
+
+            Assert.False(zspan.Tags.ContainsKey("otel.status_code")); // MUST NOT be set if the status code is UNSET.
+            Assert.False(zspan.Tags.ContainsKey("error")); // MUST NOT be set if the status code is UNSET.
```

Comment:
I could extract the `otel.status_code` and `error`. However, I have not done it because:
1. I do not think these values will ever change as they are defined in [specs](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk_exporters/zipkin.md#status)
2. Adding a const is an additional layer of indirection that decrease the readability of the test
3. I would need to make an additional test to make sure that the const value is correct